### PR TITLE
added .yaydoc.yml

### PR DESCRIPTION
**Pull request by jithinbp**
issue #107 
added a basic yaydoc.yml .
Not sure how this will work out, but it does not seem like I can set up a local build server to test yaydoc.
docs are in .md format, not .rst , so a markdown preprocessor may be needed. again, I could not locate enough examples. Simply following https://yaydoc.fossasia.org/#deployment

there's a blog post from a developer about using .md files : http://blog.fossasia.org/adding-support-for-markdown-in-yaydoc/
but it is not specified in the docs if we need to create a conf.py like the developer did, or Travis automatically handles .md files now

### CODE REVIEW

File: .yaydoc.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+metadata:
+  author: FOSSASIA
+  projectname: PSLAb-desktop-apps
+  version: development
+
+build:
+  doctheme: sphinx_fossasia_theme
+  docpath: docs
```
Comment (CloudyPadmal) [inline .yaydoc.yml]: In the https://yaydoc.fossasia.org/#deployment, this has a slash as `docs/` :)

**Comment by jithinbp** [comment]
I won't know until it is deployed.
It uses sphinx, so I'm guessing it will expect .rst files.  But at the bare minimum I'm expecting to see something that will help me proceed.

File: .yaydoc.yml (hunk 1 above)
Comment (jithinbp) [inline .yaydoc.yml]: _apps is a subdirectory in the main docs directory. It is referenced by the index file in the docs directory. similarly, there's another subdir called _utilities
Let me see how this goes.

**Comment by jithinbp** [comment]
Thanks for including the core developers of yaydoc.

I shall elaborate the current scenario for their benefit
Directory Structure of docs\
+ _config.yml :  This is the config file for Jekyll which currently generates static HTML and stores it in docs/MD_HTML
+ _apps : Contains several .md files for the numerous experiments
  + index.md : Markdown for the landing page of apps. Uses a layout that also lists all the other experiment files available
  + images : image resources for the md docs
      + schematics : SVG files . Connection diagrams for experiments
      + screenshots : PNG files . screenshots of experiment UIs
      + photographs : mostly JPG files .photographs of experiment setups
+ _utilities : Will be along the same lines as apps, but will contains docs for general purpose tools rather than experiments. The directory distinction has been made in order to automatically segregate the two without having to resort to variable declarations inside individual markdown files
+ _layouts : Contains jekyll layout files which are referenced by the md files in _apps. These are critical in deciding the look and feel of the docs
+ _includes : Include files that allow inserting utilities such as video players, sidebars, calculators etc into markdown docs with just one line of code
+ css : stylesheets for bootstrap, code highlighting etc
+ js : js for animated gif player. Others will go in here if required later

### Where do we go from here?

It has been suggested that we start using Yaydoc instead of Jekyll. I need ideas on how to get these markdown files to build with yaydoc

### Queries
Is it possible to have a local build server for yaydoc?  It's very inconvenient to get a PR accepted before one knows what the end product will look like, so I'd like to be able to use yaydoc locally before committing 


Many thanks

**Comment by sch00lb0y** [comment]
@jithinbp  you can test your repo with our hosted version http://yaydoc.herokuapp.com/. After documentation generation, you can see the preview by clicking the preview button. If you want to generate from a different branch. click `Advanced Configurations` check box to specify the branch.
 if you want to build yaydoc in your local system please follow the below step:
- install nodejs, npm, python, pip
- clone the repository
- run `npm install`
- create `.env ` file in the folder
- run `npm start`

`.env` should have following variables

GITHUB_CLIENT_ID=*****************************************************************
GITHUB_CLIENT_SECRET=*****************************************************************
GITHUB_CALLBACK_URL=http://localhost/auth/github/callback
HEROKU_CLIENT_ID=*****************************************************************
HEROKU_CLIENT_SECRET=*****************************************************************
HEROKU_CALLBACK_URL=http://localhost/auth/github/callback
SECRET=SECRET
ENCRYPTION_KEY=*****************************************************************
PORT=80
MONGODB_URI=*****************************************************************
HOSTNAME=https://e79772ba.ngrok.io
GITHUB_OAUTH_TOKEN=*****************************************************************
HEROKU_API_KEY=*****************************************************************

**Comment by sch00lb0y** [comment]
I think there is no need for writing index.rst manually for pslab dektop. yaydoc will generate automatically. if you need some advanced configuration like api blueprint then only you'll be needing manual `index.rst`.

Here is the preview of generated documentation from pslab-desktop. http://yaydoc.herokuapp.com/preview/rbalajis25@gmail.com/395531ee-ec63-4744-9881-d7d6a2a71b66_preview/ .

**Comment by jithinbp** [comment]
@sch00lb0y thank you for your help. I will try to set up a local build.

A few points 
+ The generated preview has made a list of all the links inside subdirectories in the homepage itself.
+ In addition to links to experiment pages, the toc also includes links to the images linked inside these. Defining rst files may be necessary 
+ The md headers specifying title,description,date,imagebase etc are not parsed by Yaydoc. Is there another way other than deleting the headers? Example header:
```
---
layout: expt
title: Lemon cell
date: 2017-05-21
description: experiments with a lemon cell
---
```
is rendered as
```
layout: expt title: Lemon cell date: 2017-05-21 description: experiments with a lemon cell

.. rest of the markdown is properly parsed
```

thanks

**Comment by jithinbp** [comment]
Could you also paste the contents of the yaydoc.yml file you used to generate the preview?
The preview has also rendered the contents of README.md in the base folder of the repository even though my config file mentioned `docpath:docs`

thanks

**Comment by pri22296** [comment]
@jithinbp That is the default behaviour of the auto-generated index.rst. It will include the contents of the README in the start page irrespective of the docpath.

File: .yaydoc.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+metadata:
+  author: FOSSASIA
+  projectname: PSLab-desktop-apps
+  version: development
+
+build:
+  doctheme: sphinx_fossasia_theme
+  docpath: docs
+  logo: docs/images/logo.png
```
Comment (pri22296) [inline .yaydoc.yml]: change this to images/logo.png. This path should be relative to `docpath`

**Comment by imujjwal96** [comment]
@jithinbp  Allow me to demonstrate how yaydoc will and is supposed to work. 
Have a look at https://github.com/imujjwal96/mydoc. It consists of a `docs/` directory, a `README` file and a `.yaydoc.yml` configuration file. 
Filling the relevant information in the .yaydoc.yml file, the website is generated as http://yaydoc.herokuapp.com/preview/ujjwalb1996@gmail.com/affff33e-d640-405c-b262-56616d96016d_preview/

As you can see in the docs directory, we are providing yaydoc with only the markdown files. Yaydoc generates the website from it following the hierarchy as given in the docs directory.

**Comment by imujjwal96** [comment]
Changes you need to make 
1. no need of any html, css and js
2. Store only the static images and markdown files in the docs directory having a proper hierarchy based on how you want the docs to be displayed in the generated website

**Comment by jithinbp** [comment]
html, css, js etc may be required because this was supposed to be more than just docs.
In order to set this apart from the conventional lab-manuals, it has added features such as embedded calculators for theoretical values. These require small snippets of html and js to work.

**Comment by jithinbp** [comment]
@imujjwal96 thank you for this. It clarifies a lot.
In addition to the README.md , it autogenerates a toctree based on the folders inside the docs, where md files in docs/ are listed under `contents` , while the ones inside subdirectories are listed under subheadings which share the same name as their parent directories.

The link-texts however, are derived from the H1 tags inside the respective md files.

but in case of the preview generated for pslab-desktop-apps , it is creating multiple links in the toctree for the same doc file.
psl_yay_prob

'screenshots' , 'additional reading' etc are subheadings inside the same file, and should at least appear indented
I'll get back to you after some study. 
thank you

**Comment by imujjwal96** [comment]
@jithinbp I guess that is an issue. I'll raise one at yaydoc's bug tracker. Please join the discussion there.

**Comment by imujjwal96** [comment]
@jithinbp What are md headers?
https://github.com/fossasia/pslab-desktop-apps/pull/159#issuecomment-315993986

**Comment by jithinbp** [comment]
@imujjwal96 thanks for raising the issue
by `md headers` , I meant the header section in the markdown file which looks something like this
```
---
layout: expt
title: Lemon cell
date: 2017-05-21
description: experiments with a lemon cell
---
```
The entire markdown file can be viewed here https://raw.githubusercontent.com/fossasia/pslab-desktop-apps/development/docs/_apps/B_LEMON_CELL.md

I think it is a Jekyll specific way to specify variables. The variables declared in the header section can include details such as author-name, title, description, layout to be used, custom variable etc .
These variables are readable by the template files. For example, in some md files, I merely specify a variable called `imagebase` : This is read by the template, and it automatically renders the schematic, screenshot, and photograph with the same suffix as the imagebase variable ,and adds them to the generated static page.
Since yaydoc does not appear to be as flexible yet , we may as well ignore the header section or remove them.

**Comment by pri22296** [comment]
Also I notice many `md` files which are actually `html` files. Yaydoc won't properly handle them. This does warrant a use case though of including user defined `html` or `js` files in the build docs. We'll try to add that soon.

**Comment by jithinbp** [comment]
@pri22296 The original content was in html, and I batch renamed them to md . And now I'm converting the contents to markdown sequentially, as well as adding images and such.
My Jekyll template for creating the TOC conveniently ignores the files that aren't actually in markdown  (It iterates through the `title` variables in the mdheaders ).
All files will soon be rewritten. no worries there.


As for html and js , Jekyll provides 2 ways
+ write the HTML and JS code in the md file as is. As long as the mdheader is present, they will be handled and properly rendered by Jekyll.
+ Add the code snippet to an html file created in the _includes directory , and link to it `{% include myhtmlsnippet.html %}` in your md doc.

**Pull request merged**